### PR TITLE
Enable Android Project Build when host OS is Mac

### PR DESCRIPTION
**Pull request by ggfan**
@cnorthrop @bbilodeau others, may you kindly take a look?  thanks

### CODE REVIEW

File: API-Samples/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 2.8.11)
 if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
     add_definitions(-DVK_USE_PLATFORM_WIN32_KHR -DWIN32_LEAN_AND_MEAN)
     set(DisplayServer Win32)
-elseif(CMAKE_SYSTEM_NAME STREQUAL "Android")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND ANDROID)
```
Comment (cnorthrop) [inline API-Samples/CMakeLists.txt]: I think you want to enable VK_USE_PLATFORM_ANDROID_KHR on any host OS that is targeting Android, not just Darwin.  Perhaps you could add an empty block here for Darwin, and add a new block below that checks for ANDROID on any OS.

Comment (ggfan) [inline API-Samples/CMakeLists.txt]: yeah.  added new code to get it into the correct path, please review when you get chance.  thanks

File: API-Samples/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,11 @@
 cmake_minimum_required(VERSION 2.8.11)
 
+# If ANDROID is ON, turn on cross-compiling for it
+if(ANDROID)
+    set(CMAKE_SYSTEM_NAME "Android")
+    set(CMAKE_SYSTEM_VERSION "7")
+endif()
+
```
Comment (cnorthrop) [inline API-Samples/CMakeLists.txt]: This should work for now, will need to be migrated up to root CMakeLists.txt if/when other parts of this repo add Android support.

**Comment by cnorthrop** [comment]
cherry-picked to master

**Pull request closed without merging**